src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -364,4 +377,20 @@ private static (string path, string query) RemoveTrailingIndex(string redirectio
         var (path, query, _) = UrlUtility.SplitUrl(redirectionUrl);
         return (path.EndsWith("/index", PathUtility.PathComparison) ? path[..^"index".Length] : path, query);
     }
+
+    private static bool TryGetInvalidPathChars(string path, out HashSet<char> invalidPathChars)
```

Comment:
This isn't just for redirection. `PathString` is the core type that represents a file path. We should mark a string as `PathString` in the data model if that string represents a path, we should also be checking for invalid chars for all places that turns user inputs to a `PathString`, this could be done in `PathString` json converter.

Reply:
Move the check to `PathString` json converter, not sure if we need to apply a more strict path validation here. For now, the validation will be applied on `PathString` type in redirection item and configs